src/components/input/input.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * Grow Elastic TextArea
+ *
+ * Increases the `rows` attribute of a `textarea` until it no longer scrolls.
+ *
+ * @param textarea - the target `textarea` element
+ */
+const growElasticTextArea = (textarea: HTMLTextAreaElement) => {
+  const minRows = Number(textarea.dataset.minRows);
+  let rows = Number(textarea.getAttribute('rows')) || minRows;
+  let isScrolling = textarea.scrollHeight > textarea.clientHeight;
+
+  // Grow until we stop scrolling
+  while (isScrolling) {
```

Comment:
If someone forgets to have the `c-input--elastic` class, then this causes an infinite loop that keeps increasing `rows` with no effect on the scrollbar. This also happens if it is used somewhere where there is conflicting CSS that sets `max-height` or `height`

Can we prevent the infinite loop? Here are some possibilities as to how:

- Check that rows < 500 or 1000
- Check that the height of the element has increased as a result of the `rows` attribute increasing

Reply:
Could we check for the `c-input--elastic` class when initializing?

Reply:
There is still the possibility that it could cause an infinite loop, like if that CSS did not load for some reason, or was accidently ordered with the wrong priority compared to other CSS that has fixed height

Reply:
> Could we check for the `c-input--elastic` class when initializing?

Wouldn't we be mixing styling concerns with business logic concerns if we did this? Or perhaps I'm misunderstanding. 🤔 

If the JS logic needs to depend on a CSS class, then it should be prefixed with `js-*`.

Reply:
@calebeby To be clear — you're just proposing a maximum rows value as a sanity check so the user won't cause an infinite loop. They'll still know something is wrong because the textarea will be pegged at the max value, right?

Reply:
The infinite loop happens when changing the `rows` attribute does not cause any visual change, so the scrollbar will never go away (i.e. the element has a fixed height or max-height). If there was a maximum number of rows, then if that infinite loop case occurred, then the user would see whatever the fixed height or max height is. They would not know that anything is wrong other than that the text area does not resize

Reply:
Dope. I added a maxRows check to the grow method